src/__swaps__/screens/Swap/components/SwapNumberPad.tsx

```diff
@@ -49,23 +51,78 @@ const getFormattedInputKey = (inputKey: inputKeys) => {
 
 export const SwapNumberPad = () => {
   const { isDarkMode } = useColorMode();
-  const { focusedInput, isQuoteStale, SwapInputController, configProgress, outputQuotesAreDisabled } = useSwapContext();
+  const { nativeCurrency } = useAccountSettings();
+  const {
+    focusedInput,
+    internalSelectedInputAsset,
+    internalSelectedOutputAsset,
+    isQuoteStale,
+    SwapInputController,
+    configProgress,
+    outputQuotesAreDisabled,
+  } = useSwapContext();
 
   const longPressTimer = useSharedValue(0);
 
-  const addNumber = (number?: number) => {
+  const removeFormatting = (inputKey: inputKeys) => {
     'worklet';
+    return stripNonDecimalNumbers(SwapInputController[getFormattedInputKey(inputKey)].value);
+  };
+
+  const ignoreChange = ({ currentValue, addingDecimal = false }: { currentValue?: string; addingDecimal?: boolean }) => {
+    'worklet';
+    // ignore when: outputQuotesAreDisabled and we are updating the output amount or output native value
     if ((focusedInput.value === 'outputAmount' || focusedInput.value === 'outputNativeValue') && outputQuotesAreDisabled.value) {
+      return true;
+    }
+
+    // ignore when: corresponding asset does not have a price and we are updating native inputs
+    const inputAssetPrice = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+    const outputAssetPrice = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
+    const outputAssetHasNoPrice = !outputAssetPrice || equalWorklet(outputAssetPrice, 0);
+    const inputAssetHasNoPrice = !inputAssetPrice || equalWorklet(inputAssetPrice, 0);
+    if (
+      (focusedInput.value === 'outputNativeValue' && outputAssetHasNoPrice) ||
+      (focusedInput.value === 'inputNativeValue' && inputAssetHasNoPrice)
+    ) {
+      return true;
+    }
+
+    // ignore when: decimals exceed native currency decimals
+    if (currentValue) {
+      const currentValueDecimals = currentValue.split('.')?.[1]?.length ?? -1;
```

Comment:
this section takes into account the current value and any decimals it may have, and confirms if it would exceed the allowable decimals for a native currency

Reply:
it's -1 in the instance of there not being any decimal - if there is a decimal but no numbers after, the length will be 0